### PR TITLE
Revert "Prevent noncopyable metatypes from being converted to `Any`"

### PR DESCRIPTION
**Pull request by hamishknight**
Reverts apple/swift#65606 to fix CI

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please smoke test

File: stdlib/public/Concurrency/PartialAsyncTask.swift
Comment (ktoso) [inline stdlib/public/Concurrency/PartialAsyncTask.swift:226]: Ok

**Review by ktoso — APPROVED** [review]
I guess I'm cced for the executor change of this, that's okey imho. @kavon should have a look what broke tho

**Comment by hamishknight** [comment]
Merging to unblock CI

File: stdlib/public/Concurrency/PartialAsyncTask.swift
Comment (jckarter) [reply stdlib/public/Concurrency/PartialAsyncTask.swift:226]: We can't support the use of `Self.self` as an `Any` value here yet, which is why @kavon's change correctly started flagging this as an error. Can you change that back so we can re-land @kavon's change?

Comment (kavon) [reply stdlib/public/Concurrency/PartialAsyncTask.swift:226]: I have a PR up here: https://github.com/apple/swift/pull/65668

**Pull request merged**
